sim-lib/src/lib.rs

```diff
@@ -790,13 +803,14 @@ async fn consume_events(
 /// exit if other threads signal that they have errored out.
 async fn produce_events<N: DestinationGenerator + ?Sized, A: PaymentGenerator + ?Sized>(
     source: NodeInfo,
+    activity_name: String,
     network_generator: Arc<Mutex<N>>,
     node_generator: Box<A>,
     sender: Sender<SimulationEvent>,
     shutdown: Trigger,
     listener: Listener,
 ) {
-    log::info!("Started activity producer for {source}: {node_generator}.");
+    log::info!("Started {activity_name} => activity producer for {source}: {node_generator}.");
```

Comment:
Let's have the `activity_name` as a prefix for all te logs in `produce_events` so that they're easily reference-able. Extra points if it's propagated through to other activity-relevant logging.